### ISSUE TITLE
openssl failed to build on Ubuntu 14.04.3 LTS (based on #732)

### ISSUE THREAD

**Issue by chubin** [Linuxbrew/linuxbrew#1040]
brew install openssl does not work for me:
~~~~~
47418088124064:error:25066067:DSO support routines:DLFCN_LOAD:could not load the shared library:dso_dlfcn.c:187:filename(libz.so): libz.so: cannot open shared object file: No such file or directory                                                                                                                          
47418088124064:error:25070067:DSO support routines:DSO_load:could not load the shared library:dso_lib.c:233:                                                                                                                                                                                                                   
47418088124064:error:29064065:lib(41):BIO_ZLIB_NEW:zlib not supported:c_zlib.c:463:                                                                                                                                                                                                                                            
47366714547872:error:25066067:DSO support routines:DLFCN_LOAD:could not load the shared library:dso_dlfcn.c:187:filename(libz.so): libz.so: cannot open shared object file: No such file or directory                                                                                                                          
47366714547872:error:25070067:DSO support routines:DSO_load:could not load the shared library:dso_lib.c:233:                                                                                                                                                                                                                   
47366714547872:error:29064065:lib(41):BIO_ZLIB_NEW:zlib not supported:c_zlib.c:463:  
~~~~~

Comment from Shaun Jackman:

~~~~~
I'm able to reproduce this issue on Ubuntu 14. Please open a new issue. A
workaround is

brew install gcc openssl
~~~~~

**Comment by sjackman** [comment]
I believe this issue is resolved by faf51442aba23f9498a51ae048ac008ec9b9f7ca.
Installing `openssl` works fine for me now using Ubuntu 14 and the Docker image `linuxbrew/linuxbrew`.
```sh
docker run linuxbrew/linuxbrew brew install openssl
```

**Comment by sjackman** [comment]
```
❯❯❯ docker run -it --name=linuxbrew linuxbrew/linuxbrew
linuxbrew@59d507f171f6:~$ brew install openssl
==> Installing dependencies for openssl: patchelf, makedepend, zlib, gpat
==> Installing openssl dependency: patchelf
==> Downloading https://linuxbrew.bintray.com/bottles/patchelf-0.9.x86_64_linux.
######################################################################## 100.0%
==> Pouring patchelf-0.9.x86_64_linux.bottle.tar.gz
/home/linuxbrew/.linuxbrew/Cellar/patchelf/0.9: 6 files, 838K
==> Installing openssl dependency: makedepend
==> Downloading https://linuxbrew.bintray.com/bottles/makedepend-1.0.5.x86_64_li
######################################################################## 100.0%
==> Pouring makedepend-1.0.5.x86_64_linux.bottle.tar.gz
/home/linuxbrew/.linuxbrew/Cellar/makedepend/1.0.5: 7 files, 75.4K
==> Installing openssl dependency: zlib
==> Downloading https://linuxbrew.bintray.com/bottles/zlib-1.2.8.x86_64_linux.bo
######################################################################## 100.0%
==> Pouring zlib-1.2.8.x86_64_linux.bottle.1.tar.gz
/home/linuxbrew/.linuxbrew/Cellar/zlib/1.2.8: 9 files, 370.5K
==> Installing openssl dependency: gpatch
==> Downloading https://linuxbrew.bintray.com/bottles/gpatch-2.7.5.x86_64_linux.
######################################################################## 100.0%
==> Pouring gpatch-2.7.5.x86_64_linux.bottle.1.tar.gz
/home/linuxbrew/.linuxbrew/Cellar/gpatch/2.7.5: 9 files, 293.5K
==> Installing openssl
==> Downloading https://www.openssl.org/source/openssl-1.0.2g.tar.gz
######################################################################## 100.0%
==> Downloading https://raw.githubusercontent.com/Homebrew/patches/3f1dc8ea145a7
######################################################################## 100.0%
==> Downloading https://raw.githubusercontent.com/Homebrew/patches/3f1dc8ea145a7
######################################################################## 100.0%
==> Patching
==> Applying revert-pass-pure-constants-verbatim.patch
patching file crypto/perlasm/x86_64-xlate.pl
==> Applying tshort-asm.patch
patching file crypto/perlasm/x86_64-xlate.pl
==> perl ./Configure --prefix=/home/linuxbrew/.linuxbrew/Cellar/openssl/1.0.2g -
==> make depend
==> make
==> make test
==> make install MANDIR=/home/linuxbrew/.linuxbrew/Cellar/openssl/1.0.2g/share/m
==> Downloading https://raw.githubusercontent.com/bagder/ca-bundle/bff056d04b9e2
######################################################################## 100.0%
==> Caveats
A CA file has been bootstrapped using certificates from the system
keychain. To add additional certificates, place .pem files in
  /home/linuxbrew/.linuxbrew/etc/openssl/certs

and run
  /home/linuxbrew/.linuxbrew/opt/openssl/bin/c_rehash

==> Summary
/home/linuxbrew/.linuxbrew/Cellar/openssl/1.0.2g: 466 files, 14.2M, built in 2 minutes 36 seconds
```
